presto-orc/src/main/java/com/facebook/presto/orc/OrcWriter.java

```diff
@@ -328,22 +328,29 @@ private void flushStripe(FlushReason flushReason)
             outputData.add(createDataOutput(MAGIC));
             stripeStartOffset += MAGIC.length();
         }
-        // add stripe data
-        outputData.addAll(bufferStripeData(stripeStartOffset, flushReason));
-        // if the file is being closed, add the file footer
-        if (flushReason == CLOSED) {
-            outputData.addAll(bufferFileFooter());
-        }
 
-        // write all data
-        orcDataSink.write(outputData);
+        try {
+            // add stripe data
+            outputData.addAll(bufferStripeData(stripeStartOffset, flushReason));
+            // if the file is being closed, add the file footer
+            if (flushReason == CLOSED) {
+                outputData.addAll(bufferFileFooter());
+            }
 
-        // open next stripe
-        columnWriters.forEach(ColumnWriter::reset);
-        dictionaryCompressionOptimizer.reset();
-        rowGroupRowCount = 0;
-        stripeRowCount = 0;
-        bufferedBytes = toIntExact(columnWriters.stream().mapToLong(ColumnWriter::getBufferedBytes).sum());
+            // write all data
+            orcDataSink.write(outputData);
+        }
+        catch (IOException e) {
```

Comment:
This `catch` is redundant. Without it, the code should be able to throw `e` as well